### PR TITLE
Accept plain text/csv upload requests

### PR DESCRIPTION
**Pull request by villaaston**
Extended to use the request body as a file for text/csv content types.

Some systems (looking at you Workday) can't create a valid multipart request (according to the RFC Flask uses), so we have to accept text/plain OR text/csv requests.

Using the body as FileStorage means that the rest of the interfaces don't need to change.

Also tidied up a few other bits and bobs.

### CODE REVIEW

**Comment by villaaston** [comment]
Tests coming in a separate PR (the test file uses two spaces for indentation, so it's messy)

File: flask_gae_gcs.py [hunk 1]
```diff
@@ -218,28 +216,45 @@ def save_files(fields, validators=None, retry_params=None, bucket_name=None):
...
     result = []
-    for key, value in request.files.iteritems():
-        if not isinstance(value, unicode):
-            result.append((key, value))
+
+    content_type = request.headers.get('content-type')
+    mime_type, _ = parse_header(content_type)
+
+    if mime_type in ('text/plain', 'text/csv'):
+        _, params = parse_header(request.headers.get('content-disposition'))
```
Comment (obmarg) [inline flask_gae_gcs.py]: The docstring suggests that content-disposition is optional, but I assume this'll fail if it's not there.

Comment (villaaston) [inline flask_gae_gcs.py]: 't will indeed

**Comment by villaaston** [comment]
It's really hard to get the tests running for this project - will hit up @siame when he's back and see how he got them running. For now this has been tested a lot locally and on QA and works well. Anyone against a merge and release of the new version?

**Pull request merged**